tap_github/user_streams.py

```diff
@@ -74,3 +74,86 @@ def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
         th.Property("updated_at", th.DateTimeType),
         th.Property("created_at", th.DateTimeType),
     ).to_dict()
+
+
+class StarredStream(GitHubStream):
+    """Defines 'Stars' stream. Warning: this stream does NOT track star deletions."""
+
+    name = "starred"
+    path = "/users/{username}/starred"
+    # "repo_id" is the starred repo's id.
+    primary_keys = ["repo_id", "username"]
+    parent_stream_type = UserStream
+    state_partitioning_keys = ["username"]
+    replication_key = "starred_at"
+
+    @property
+    def http_headers(self) -> dict:
+        """Return the http headers needed.
+
+        Overridden to use an endpoint which includes starred_at property:
+        https://docs.github.com/en/rest/reference/activity#custom-media-types-for-starring
+        """
+        headers = super().http_headers
+        headers["Accept"] = "application/vnd.github.v3.star+json"
+        return headers
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        """
+        Add a repo_id top-level field to be used as state replication key.
+        """
+        row["repo_id"] = row["repo"]["id"]
+        return row
+
+    schema = th.PropertiesList(
+        # Parent Keys
+        th.Property("username", th.StringType),
+        th.Property("repo_id", th.StringType),
+        # Starred Repo Info
+        th.Property("starred_at", th.DateTimeType),
+        th.Property(
+            "repo",
+            th.ObjectType(
+                th.Property("id", th.IntegerType),
+                th.Property("node_id", th.StringType),
+                th.Property("full_name", th.StringType),
+                th.Property("description", th.StringType),
+                th.Property("html_url", th.StringType),
+                th.Property(
+                    "owner",
+                    th.ObjectType(
+                        th.Property("login", th.StringType),
+                        th.Property("id", th.IntegerType),
+                        th.Property("node_id", th.StringType),
+                        th.Property("type", th.StringType),
+                        th.Property("avatar_url", th.StringType),
+                        th.Property("html_url", th.StringType),
+                        th.Property("site_admin", th.BooleanType),
+                    ),
```

Comment:
Probably outside of scope for this PR, but this schema are reused [in a few places](https://github.com/MeltanoLabs/tap-github/blob/4b6d7ca9ba604739b1ee3615acd84ea7c5c67b7b/tap_github/repository_streams.py#L89-L98), so perhaps an `OWNER_SCHEMA` can be refactored out of this?

Other schemas are likely reused too.

Reply:
Yeah I agree, it would be nice to have a file with shared schemas. I'll open an issue.